core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -92,7 +92,7 @@ public GenericManifestFile(Schema avroSchema) {
     this.specId = specId;
     this.sequenceNumber = 0;
     this.minSequenceNumber = 0;
-    this.snapshotId = null;
+    this.snapshotId = 0L;
```

Comment:
This seems a bit dangerous to me. I get we're saying it's required, so we don't need the null value but I looked around to see where manifestFile.snapshotId() == null checks are performed, and there's quite a few. For example `RewriteDataFiles`, `AppendFiles`. This may break some of the existing logic there by exercising the other branch in the code?

Reply:
Hmm, I thought that this was used only by tests, but there is one path from actual code that also uses this. Let me check.

Reply:
This constructor is used when there is no manifest-list (v1 tables). That's quite rare nowadays.

Reply:
Hey @amogh-jahagirdar I looked at it once more, and it looks like the Snapshot-ID is available, so no need in setting it to `0L`. Thanks for pointing this out